optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/common/AbstractJoiner.java

```diff
@@ -16,8 +16,30 @@
 
 package org.optaplanner.core.impl.score.stream.common;
 
-public abstract class AbstractJoiner {
+import java.util.Arrays;
+import java.util.function.Function;
+import java.util.stream.IntStream;
+
+public abstract class AbstractJoiner<Right> {
```

Comment:
This presumes that that we'll never do this:
`Bi<A,B>.join(Bi<C,D>)`

-1 to abstract the Right generatic type here...